### ISSUE TITLE
Forgot Password UX Updates

### ISSUE THREAD

**Issue by austintruchan** [matrix-msu/Kora3#297]
- [ ] We need to slightly update the password reset page the user is greeted with after clicking the update password button in the forgot password email. I think mainly updating the header size, and moving the form inputs down a bit away from the header.
This page is here: https://projects.invisionapp.com/d/main#/console/11107725/259280106/preview

- [ ] It would be pretty awesome if the email field was already filled in for the user on this page.

After the user fills out their new password on this page, it seems to auto log them in as "non admin" user. I think this happens when the user creates a new password for an email that doesn't exist? 
<img width="1610" alt="screen shot 2018-09-11 at 1 23 58 pm" src="https://user-images.githubusercontent.com/5408529/45376554-18430e80-b5c6-11e8-936d-bd5946d85e60.png">

- [ ] We could fix this situation, of adding in an email that doesn't exist, by stopping the forgot password to send an email altogether if that email doesn't exist already within k3. We need to add an error state in the forgot password modal, for "Email does not exist", and not allow the user to submit the modal unless they put in an email that already exists in k3.

- [ ] Add in a static, non disappearing notification after the user hits the "Send Password Reset Link" button in the forgot password modal. This notification looks like this: https://projects.invisionapp.com/d/main#/console/11107725/259288984/preview
@suuuth can help with notification set up.

- [ ] Need a disappearing notification after the user successfully updates their password and is logged in. This Notification can be the same as the other "Password Successfully Reset!" notifications.

**Comment by Dev34** [comment]
@austintruchan Can you see if you can still reproduce the non-admin login bug you detailed? When I try to use the Forgot Password modal with a non-existent password, or I try to enter a new password for a non-existent email, I get the 'We can't find a user with that e-mail address.' validation error and the request doesn't go through.

**Comment by Dev34** [comment]
@austintruchan 
Can I get rid of the old notification design entirely then?
Also emails are now working on kora3 devs again.

**Comment by austintruchan** [comment]
@Dev34 Yes you can. And okay, let me try again.

**Comment by Dev34** [comment]
Removed the old notification styling. Looks like the comp now. This affects all notifications on the welcome page.

Stopped the Forgot Password modal from closing when they enter an email that doesn't exist in the database.

This is on the forgot-password-#297 branch (pushed to GitHub)

**Comment by austintruchan** [comment]
- [ ] Text in notification should be left aligned
<img width="435" alt="screen shot 2018-10-03 at 2 48 35 pm" src="https://user-images.githubusercontent.com/5408529/46432179-9903af00-c71b-11e8-9859-2aeac8b3799e.png">

- [ ] On mobile, I would stack and center align the notification text. So that "Check your email" is on top of the rest of the text.
<img width="395" alt="screen shot 2018-10-03 at 2 53 11 pm" src="https://user-images.githubusercontent.com/5408529/46432393-26df9a00-c71c-11e8-965f-7ed111aecac5.png">


- [ ] I unfortunately am not receiving emails for my austintruchan@gmail.com account or my truchan@msu.edu account, so I can't test the other pages. Is there any other way I can test?

**Comment by Dev34** [comment]
The notifications on the welcome page should now be styled correctly: left-aligned, and center-aligned with stacking for mobile. Make sure to run ```compass compile```

Hmm. Your email config settings might not be correct. You could try verifying that the MAIL_ variables in the .env file match what Anthony recently posted in the kora3-programming channel. 
Alternatively, after requesting a password reset email, you can go to the log file at Kora3/storage/logs and you'll see part of a URL there which contains your password reset token. If you fill out that URL and visit it, you can also reach the password reset page. Added this just for testing purposes.

This is on the forgot-password-#297 branch (pushed to GitHub)

**Comment by austintruchan** [comment]
Ok. I was able to get it working.

- [ ] For the reset password page, remove the padding top of 100px that is above the header.

- [ ] Adjust these inputs so that the behave appropriately for mobile, and have correct margins. Model after the main login page on mobile.

- [ ] After resetting my password, it redirected me to a page that looked like this: 
<img width="1512" alt="screen shot 2018-10-09 at 11 58 05 am" src="https://user-images.githubusercontent.com/5408529/46682081-a1943380-cbba-11e8-98f6-307658895c80.png">

**Comment by Dev34** [comment]
Removed the 100px padding above the header.
Added mobile responsiveness to the inputs just like on the main login page.
I couldn't reproduce that redirect bug, but I think I fixed it.

This is on the forgot-password-#297 branch (pushed to GitHub)

**Comment by austintruchan** [comment]
- [ ] There is a noticeable pause between the time the loading animation ends, and the forgot password modal closes after entering in an acceptable email. Anyway we could keep the loading animation around until the modal closes out completely and the notification shows up?

- [ ] The redirect bug is fixed, but I am being redirected to the "Thanks for signing up screen". I should be redirected to the regular login screen.

**Comment by Dev34** [comment]
Loading animation now stays around until redirect. Make sure to run ```compass compile```.

I'm dumbfounded by the redirect issue.. It should be redirecting to the main projects page after a password reset (which it does on my dev on Edge & Chrome). It's very explicitly handled by ```return redirect()->action('ProjectController@index')->with('k3_global_success', 'password_reset');``` Do you mind verifying one more time that this bug exists, and then I'll see who I can talk to about diagnosing and fixing it?

This is on the forgot-password-#297 branch (pushed to GitHub)

**Comment by austintruchan** [comment]
Loading animation looks good.

Tested again and I am still being directed to the thanks for signing up page. Tested on Safari and Chrome for Mac.

**Comment by Dev34** [comment]
I was able to reproduce getting redirected to the 'thanks for signing up page', by creating an account and immediately trying to reset the password (without first confirming the account via email with the authentication token). I assume this is what you were doing? If not, then it is likely that the 'active' boolean associated with your kora3 account was inadvertently flipped to false. That can be fixed by flipping that bit in the kora3_users database table with MySQL Workbench.

I added an error message to the Forgot Password modal on the welcome page for when an email is entered that hasn't been activated. This stops the modal from closing.

This is on the forgot-password-#297 branch (pushed to GitHub)

**Comment by austintruchan** [comment]
Nope, was trying this on previously created accounts that I've used before, not on new accounts. Now there is no way for me to reset my password on these new accounts, nor can I even login with them, as it takes me to the thanks for signing up page upon login, even though these accounts have already been created, and authenticated before. I will try testing on a brand new account and see what happens when I try resettting my password after authenticating.

**Comment by Dev34** [comment]
I have no idea what could be causing an issue like that.
I updated the error message for unactivated accounts in Forgot Password modal on welcome page, so that it is more accurate.

This is on the forgot-password-#297 branch (pushed to GitHub)

**Comment by austintruchan** [comment]
Okay, yeah let me do further testing. Theres something up with my dev, anthony is currently looking into it, and then I'll test some more.

**Comment by austintruchan** [comment]
Okay! I figured it out. For whatever reason the accounts that I was using were set to Inactive in the user management page. I switched them to active and now it works as it should.
